vars/mtaBuild.groovy

```diff
@@ -75,106 +53,53 @@ void call(Map parameters = [:]) {
             .dependingOn('mtaBuildTool').mixin('dockerImage')
             .use()
 
-        new Utils().pushToSWA([
-            step: STEP_NAME,
-            stepParamKey1: 'scriptMissing',
-            stepParam1: parameters?.script == null
-        ], configuration)
-
-        dockerExecute(
-            script: script,
-            dockerImage: configuration.dockerImage,
-            dockerEnvVars: configuration.dockerEnvVars,
-            dockerOptions: configuration.dockerOptions,
-            dockerWorkspace: configuration.dockerWorkspace
-        ) {
-
-            String projectSettingsFile = configuration.projectSettingsFile?.trim()
-            if (projectSettingsFile) {
-                if (projectSettingsFile.startsWith("http")) {
-                    projectSettingsFile = downloadSettingsFromUrl(this, projectSettingsFile, 'project-settings.xml')
-                }
-                sh 'mkdir -p $HOME/.m2'
-                sh "cp ${projectSettingsFile} \$HOME/.m2/settings.xml"
-            }
-
-            String globalSettingsFile = configuration.globalSettingsFile?.trim()
-            if (globalSettingsFile) {
-                if (globalSettingsFile.startsWith("http")) {
-                    globalSettingsFile = downloadSettingsFromUrl(this, globalSettingsFile, 'global-settings.xml')
-                }
-                sh "cp ${globalSettingsFile} \$M2_HOME/conf/settings.xml"
-            }
+        String configFiles = piperGoUtils.prepareConfigurations([PIPER_DEFAULTS].plus(script.commonPipelineEnvironment.getCustomDefaults()), ADDITIONAL_CONFIGS_FOLDER)
+        writeFile(file: "${METADATA_FOLDER}/${METADATA_FILE}", text: libraryResource(METADATA_FILE))
+        Map contextConfig = readJSON(text: sh(returnStdout: true, script: "./piper getConfig --stepMetadata '${METADATA_FOLDER}/${METADATA_FILE}' --defaultConfig ${configFiles} --contextConfig"))
 
-            String defaultNpmRegistry = configuration.defaultNpmRegistry?.trim()
-            if (defaultNpmRegistry) {
-                sh "npm config set registry $defaultNpmRegistry"
-            }
+        parameters = [:] << parameters
```

Comment:
This looks strange. What is achieved by this line?

Reply:
A new parameter map is created and initialized with the current parameters. After that the new parameter are modified. The old parameter map as it was handed in into the step via signature is not modified (some kind of "defensive copy".

Reply:
Ok, the intention is sound. I've put this into a small test, since I wanted to learn whether that actually makes a deep copy:
```
 Map parameters = [:]
parameters.list = ['apple', 'oranges', 'peas']

Map copiedParameters = [:] << parameters
copiedParameters.list.add('salad')

System.out.println('parameters.list: ' + parameters.list)
```
Output is:
```
parameters.list: [apple, oranges, peas, salad]
```
Note that some Groovy operators behave in unexpected ways, for example `+=` generates a copy of the object it is appending to, and assigns that new copy to whatever is on the left of the operator. So `copiedParameters.list += 'salad'` would have worked. But only by "accident", the root problem is that you are not actually making a deep-copy of the parameters as you intended.

Reply:
We can drop that and use the parameters map ...

It does not serve as a real "defensive" copy due to the deep copy behaour I am aware of. But later on I only manipulate on the root level of the parameter map. Hence for the purposes here is is sufficient like it is ...

But again: in case you think we should not blow up the code like this I can drop that ...

Reply:
No, I just wanted to understand. If you manipulate on the root-level only, (I saw that you remove keys), then it achieves the purpose.